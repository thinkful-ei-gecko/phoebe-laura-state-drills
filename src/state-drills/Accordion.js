import React from "react";

class Accordion extends React.Component {
  static defaultProps = { sections : [] };


  render() {
    const buttons = this.defaultProps.sections.map((section, index) => (
      <button key={index}> {section.title} </button>
    ));
    return <ul>
      <li>
        {this.buttons}
        <p>Section content</p>
      </li>
    </ul>
  }
}

export default Accordion;