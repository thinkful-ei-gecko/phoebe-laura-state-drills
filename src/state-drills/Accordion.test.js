import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import renderer from 'react-test-renderer';

const sections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
  ]

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Accordion sections={sections} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<Accordion sections={sections} />)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });
  