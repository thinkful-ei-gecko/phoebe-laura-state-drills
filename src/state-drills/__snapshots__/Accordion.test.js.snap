// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<ul>
  <li>
    <p>
      Section content
    </p>
  </li>
</ul>
`;
